src/NuGet.Core/Microsoft.Build.NuGetSdkResolver/GlobalJsonReader.cs

```diff
@@ -19,7 +17,9 @@ internal static class GlobalJsonReader
     {
         public const string GlobalJsonFileName = "global.json";
 
-        public const string MSBuildSdksPropertyName = "msbuild-sdks";
+        private static readonly Regex MsBuildSdksSectionRegex = new Regex("\\\"msbuild-sdks\\\"\\s*:\\s*\\{(?<Packages>[^}]+)\\}");
```

Comment:
Please add a timeout.

Reply:
Do you have a suggested timeout value?

Reply:
Considering the size of the global.json file I wouldn't add anything more than a 1-2s. 

The general idea is to have a timeout in the first place.

Reply:
I totally agree and think 1 second should be fine since its taking 29ms on my machine